### ISSUE TITLE
GraphQL Codegen does not work with ES Modules

### ISSUE THREAD

**Issue by Hexatare** [Shopify/shopify-app-js#1738]
# Issue summary

The GraphQL Codegen creates two files: `admin.types.d.ts` and `admin.generated.d.ts`. The codegen preset automatically adds an import statement at the top of the `admin.generated.d.ts` file to include the required base types. However, when using `"type": "module"` in the `package.json`, the import statement is invalid because you are required to specify the `.js` file extension. There is currently no way to configure this behavior to allow the codegen to generate the correct import path with the `.js` extension. This means that I have to manually update the generated files after each codegen run or copy the code from the repository.

## Expected behavior

There should either be an option to update the import or the codegen should detect when it is used inside an ESM module.

## Actual behavior

The import statement in the generated `admin.generated.d.ts` file does not include the required `.js` extension when using `"type": "module"` in `package.json`, resulting in an invalid import. This forces developers to either manually edit the file after each codegen run or maintain a custom codegen setup.

## Steps to reproduce the problem

1. Set `"type": "module"` in `package.json`.
2. Run the GraphQL codegen.
3. Inspect the generated `admin.generated.d.ts` file.
4. Notice the incorrect import statement that doesn't include the `.js` extension, causing issues when trying to import from `admin.types`.

## Debug logs
The generated output looks like this:
```ts
/* eslint-disable eslint-comments/disable-enable-pair */
/* eslint-disable eslint-comments/no-unlimited-disable */
/* eslint-disable */
import type * as AdminTypes from './admin.types';

...
```

**Comment by petertoth** [comment]
I am experiencing the same issue. I tried passing [emitLegacyCommonJSImports](https://the-guild.dev/graphql/codegen/docs/getting-started/esm-typescript-usage) option as `emitLegacyCommonJSImports: false` to my `.graphql.rc` config but without any luck

**Comment by Hexatare** [comment]
Yeah, I tried that too. The problem is that since the import statement is not created by the graphql-codegen package but the custom Shopify layer on top, it doesn't have an effect.

**Comment by mateboer-seon** [comment]
Any updates on this?

**Comment by florianbepunkt** [comment]
@lizkenyon It would be really great if this could be fixed by the Shopify team. This currently defeats the point of the package for ESM users – **automatically** creating the types.

**Comment by florianbepunkt** [comment]
Not stale.

**Comment by faizanabidnaqvi** [comment]
We are facing the same issue. I thought following the guide [here](https://the-guild.dev/graphql/codegen/docs/getting-started/esm-typescript-usage) would work, but it does not. We spent several hours trying to find a workaround but like said above it is generated by shopify so we can't control it. 

For now, we wrote a script to run every time after codegen is run.

```
import fs from 'node:fs';
const file = './src/types/shopify-gql/admin.generated.ts';

let content = fs.readFileSync(file, 'utf8');
content = content.replaceAll(
    "from './admin.types'",
    "from './admin.types.js'"
);
fs.writeFileSync(file, content);
console.log('Patched admin.generated.ts import to .js');
```

**Comment by byrichardpowell** [comment]
Hey everyone 👋 

This should have been fixed in the latest release.  As such, I'm going to close the issue.
